Qa links learn more and link for fees in footer

sedatbsr opened the pull request:
### Description

Test case for Settings - _**Country**_ - Onboarding - Links Learn more and link for fees in footer added.
* Data file created under _.test-data_` learn-more-links.data.ts`
* A new .spec file has been created under _dashboard-ui_ `learn-more-links.spec.ts`
* Unused imports removed from .spec files
* A locator and functions have been to _pcp-onboarding_ POM file for the new test case.

Commit:
Test case added

-Links Learn more and link for fees in footer per country

Commit:
fix: Test key added

Commit:
fix: unused imports removed

mishautkin commented on tests/qa/utils/admin/pcp-onboarding.ts:
```diff
@@ -108,5 +110,49 @@ export class PcpOnboarding extends PcpAdminPage {
+	getLearnMoreLinks = async (): Promise<
+		{ url: string; title: string }[]
+	> => {
+		await this.page.waitForSelector( '.ppcp-r-welcome-docs a', {
```
I see that you have the locator created. Please add locator and use `.waitFor()`:
```ts
wait this.page.learnMoreLinks().waitFor( { state: 'visible' } );
```

sedatbsr replied:
I tried this but because of the PW strict mode, it fails. Because it returns more than one element. I can remove the locator and use it directly inside the function.

sedatbsr replied:
Alright. If I wait for the `.first()` one, it works.

mishautkin commented on tests/qa/utils/admin/pcp-onboarding.ts:
```diff
@@ -108,5 +110,49 @@ export class PcpOnboarding extends PcpAdminPage {
+				} );
+			}
+		}
+		console.log( 'extractedLinks:', extractedLinks );
```
Remove `console.log`

mishautkin commented on tests/qa/utils/admin/pcp-onboarding.ts:
```diff
@@ -108,5 +110,49 @@ export class PcpOnboarding extends PcpAdminPage {
+		return newPage;
+	};
+
+	getPageTitle = async ( newPage = this.page ) => {
```
This is just a one-line method. May be you may directly use `await newPage.title();` in a test?

mishautkin commented on tests/qa/tests/02-dashboard-ui/learn-more-links.spec.ts:
```diff
@@ -0,0 +1,45 @@
+
+			await pcpOnboarding.visit();
+			await pcpOnboarding.page.waitForLoadState( 'networkidle' );
+			const pageLinks = await pcpOnboarding.getLearnMoreLinks();
```
I hope that I understood the purpose of `getLearnMoreLinks` correctly. I would suggest to use soft assertions instead.

In this case you don't need to use links array comparison and can avoid additional methods in `pcpOnboarding`, unless there are more tests to use same functionality. I give example of my suggested test version in one code block (but it can be split into methods if to be reused):
```ts
await pcpOnboarding.visit();
await pcpOnboarding.waitForLoadingMaskRemoved();
for ( const { url, title } of expectedLinks ) {
    const link = this.pcpOnboarding.learnMoreLinks().locator( `[href="${ url }"]` );
    // Assert that expected link is visible but do not stop test if not
    await expect.soft( link ).toBeVisible();
    // If link is visible
    if (await link.isVisible()) {
        // Open link in a new context to not revisit Onboarding again
	const newContext = await this.page.context().browser()?.newContext();
        const newPage = await newContext?.newPage();
        await newPage?.goto(link);
	// Assertions
	expect.soft( newPage.url() ).toBe( url );
	expect.soft( await newPage.title() ).toContain( title );
        // Close new page and context
	await newPage.close();
	await newContext.close();
    }
}

sedatbsr replied:
So you mean that there is no need to create a function and directly write the code test block, right? I didn't think of soft assertion's usage which sounds reasonable.

mishautkin replied:
I'd say, Having these methods in `pcpOnboarding` is reasonable if you're going to reuse them more then once across several specs.

If it's only used in one test - you can place code in a test directly or create a function in a spec file, if the code is bulky (or used in more than one test of a spec).

If "Learn more" links are going to be tested across several fixtures then methods could be placed in `pcp-admin-page.ts`. This would allow all pcp-fixtures to inherit them.

sedatbsr replied:
It didn't work no matter how much I tried. It fails when it comes to `await expect.soft( link ).toBeVisible();` 
Here is what I did:
* I checked the locators and printed in the console to be sure what it returns.
* Tried a the waits including `waitForLoadingMaskRemoved`. 
* Used timeout in locator.
* waited before soft assertion
* Checked if the hrefs are dynamically loaded

Commit:
Test updated

Functions removed from pcp-onboarding
Logic placed in spec file.
Data file re-organized.

Commit:
fix: .only removed
lint

Commit:
fix: unused methods and selectors removed

mishautkin commented on tests/qa/tests/02-dashboard-ui/learn-more-links.spec.ts:
```diff
@@ -11,34 +11,34 @@ test.describe( () => {
 
 	for ( const country of countries ) {
-		test( `PCP-0000 | Settings -  ${ country } - Onboarding - Links Learn more and link for fees in footer`, async ( {
+		test.only( learnMoreLinksByCountry[country].testTitle, async ( {
```
Remove `.only`, please.

mishautkin reviewed (commented):
A minor comment. Also I'm not sure if the initial methods are present in `pcpOnboarding`. You may stash them locally but please check if they're not in the commit if they're not used.
